lib/smack/Prelude.cpp

```diff
@@ -280,6 +319,9 @@ struct IntOpGen::IntArithOp : public IntOp {
       : IntOp(opName, arity, intOp, bvOp, alsoUsedByPtr) {}
 
   FuncDecl *getIntFunc(unsigned size) const {
+    if (!intOp)
+      return nullptr;
```

Comment:
Are callers of getIntFunc ready to deal with the possibility of a null value here? Might be better to either use [the optional type](https://en.cppreference.com/w/cpp/utility/optional) or raise an error.

Reply:
Yes, it can deal with null values. I prefer the current pointer version otherwise I need to change a lot of code.